src/api/app/models/project.rb

```diff
@@ -1455,6 +1455,15 @@ def self.parent_projects(project_name)
     projects
   end
 
+  def lock(comment = nil)
+    self.transaction do
+      f = self.flags.find_by_flag_and_status('lock', 'disable')
```

Comment:
2 SQL calls instead of 4 :-)

```ruby
def lock(comment = nil)
  unlocked_flag = self.flags.find_by_flag_and_status('lock', 'disable')
  if unlocked_flag
    unlocked_flag.status = enable
  else
    self.flags.build(flag: 'lock', status: 'enable')
  end
  self.store({comment: comment})
end
```